### DISCUSSION TITLE
<title>Python vs Java which is best for web development </title>

### DISCUSSION THREAD

**Discussion by IshikaMahajan7** [community/community#41784] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

![pythonvsjava_cover](https://user-images.githubusercontent.com/115925848/207746737-dbf8353b-bf9c-46d8-9f0e-2ef5ee944f56.png)

which one is best for web development?

**Comment by sagarbangade** [comment, marked as answer]
Java is better developed for building web applications.

**Comment by sadman-shami** [comment]
I think python will be easy for you. Even Web development in python is far better than java (personal preference). You can do machine learning stuffs in python.

**Comment by jfullstackdev** [comment]
programming languages were created due to a specific need, intentional or not, and it will be unfair to compare them to others when they were built for a different purpose

one will say, "C Language is dead" then you came across Linux development and you realize it's still being used today because it's for system programming

there are dying languages because one, it's not the preferred language anymore being replaced by another language, two, when it's discontinued

and trending languages are there not because they are just trending, they are the selected programming language because of practical reasons

**Comment by eleohoang** [comment]
Maybe you like free style then you will like Python. But other programmers who like everything tightly controlled will like Java. And if you're new to programming, it's better to begin with Python just because it's easy to learn.
